ocaml/xapi/xapi_pool.ml

```diff
@@ -908,24 +908,40 @@ let eject ~__context ~host =
     in
 
     let write_first_boot_management_interface_configuration_file () =
+      (* During pool.eject firstboot scripts running on the ejected host will write the new
+       * management_interface in inventory. In case of VLAN - new vlan bridge will be written. *)
       let bridge = Xapi_pif.bridge_naming_convention management_device in
       Xapi_inventory.update Xapi_inventory._management_interface bridge;
       let primary_address_type = Db.PIF.get_primary_address_type ~__context ~self:management_pif in
       Xapi_inventory.update Xapi_inventory._management_address_type
         (Record_util.primary_address_type_to_string primary_address_type);
-      let configuration_file_contents = begin
-        "LABEL='" ^ management_device ^ "'\nMODE=" ^ mode ^
-        if mode = "static" then
-          "\nIP=" ^ pif.API.pIF_IP ^
-          "\nNETMASK=" ^ pif.API.pIF_netmask ^
-          "\nGATEWAY=" ^ pif.API.pIF_gateway ^
-          "\nDNS=" ^ pif.API.pIF_DNS ^ "\n"
-        else
-          "\n"
-      end in
+      let sprintf = Printf.sprintf in
+      (* If the management_interface exists on a vlan, write the vlan id into management.conf *)
+      let vlan_id = Int64.to_int pif.API.pIF_VLAN in
+      let config_base =
+        [ sprintf "LABEL='%s'" management_device
+        ; sprintf "MODE=%s" mode
+        ] in
+      let config_static = if mode <> "static" then [] else
+        [ sprintf "IP=%s" pif.API.pIF_IP
+        ; sprintf "NETMASK=%s" pif.API.pIF_netmask
+        ; sprintf "GATEWAY=%s" pif.API.pIF_gateway
+        ; sprintf "DNS=%s" pif.API.pIF_DNS
+        ] in
+      let config_vlan = if vlan_id = -1 then [] else
+        [ sprintf "VLAN=%d" vlan_id
+        ] in
+      let configuration_file =
+        List.concat
+          [ config_base
+          ; config_static
+          ; config_vlan
+          ]
+          |> String.concat "\n"
+      in
       Unixext.write_string_to_file
         (Xapi_globs.first_boot_dir ^ "data/management.conf")
-        (configuration_file_contents) in
+        (configuration_file) in
```

Comment:
You don't need the parentheses here but I would not mind. Otherwise this looks good to go.